include/nonius/benchmark.h++

```diff
@@ -63,7 +64,11 @@ namespace nonius {
             times.reserve(cfg.samples);
             std::generate_n(std::back_inserter(times), cfg.samples, [this, env, plan]{
                     detail::chronometer_model<Clock> model;
+
+                    detail::optimizer_barrier();
                     (*this)(chronometer(model, plan.iterations_per_sample));
+                    detail::optimizer_barrier();
```

Comment:
I would suggest putting the `optimizer_barrier` around the function call to the concrete benchmark, instead of the whole measurement. That way we can also use the return value for the deoptimization, which would make the trick compatible with MSVC and with what the documentation suggests.

```
template <typename T>
void deoptimize_value(T&& x) { 
    keep_memory(&x); 
}

template <typename Fn, typename... Args>
auto invoke_deoptimized(Fn&& fn, Args&&... args) -> std::enable_if_t<!std::is_same<void, std::result_of_t<Fn(Args...)>>{}> {
    deoptimize_value(std::forward<Fn>(fn) (std::forward<Args>(args...)));
}

template <typename Fn, typename... Args>
auto invoke_deoptimized(Fn&& fn, Args&&... args) -> std::enable_if_t<std::is_same<void, std::result_of_t<Fn(Args...)>>{}> {
    std::forward<Fn>(fn) (std::forward<Args>(args...));
    // maybe call optimizer_barrier() ??
}
```

And then in `chronometer::measure`:
```

        template <typename Fun>
        void measure(Fun&& fun, std::false_type) {
            measure([&fun](int) { return fun(); }); // added return !
        }

        template <typename Fun>
        void measure(Fun&& fun, std::true_type) {
            impl->start();
            for(int i = 0; i < k; ++i) 
                deoptimized_invoke(fun, i);
            impl->finish();
        }
```

I did not try the code and it is C++14, but you get the idea... What do you think?